Cpp_test/lib/ipc-shm-send.cpp

```diff
@@ -0,0 +1,115 @@
+#include "ipc-shm-send.h"
+#include <iostream>
+#include <cassert>
+
+IPCShmSend::~IPCShmSend()
+{
+	
+	shm_unlink(this->info.method_name);
+}
+
+void IPCShmSend::init()
+{
+	this->file_handler.open_file();
+	this->open_shm();
+	this->set_shm_size();
+	this->map_shm();
+	if (this->shmd > 0)
+	{
+		close(this->shmd);
+	}
+	pthread_mutexattr_destroy(&(this->mutex_attr));
```

Comment:
I think (but it's only my opinion) that destroying the mutexattr and the condattr should be in init_mutex() / init_cond() methods